integration-tests/src/integrationTest/java/pl/allegro/tech/hermes/integrationtests/slow/KafkaReadinessCheckTest.java

```diff
@@ -0,0 +1,203 @@
+        Stream.of(hermesZookeeper, kafka).parallel().forEach(Startable::stop);
+    }
+
+    @Test
+    public void shouldNotBeReadyUntilKafkaClusterIsUp() {
+        // given
+        kafka.cutOffConnectionsBetweenBrokersAndClients();
+
+        // when
+        HermesTestApp hermesFrontend = new HermesFrontendTestApp(hermesZookeeper, kafka)
+                .metadataMaxAgeInSeconds(1)
+                .readinessCheckIntervalInSeconds(1)
+                .kafkaCheckEnabled()
+                .start();
+
+        // then
+        getStatusReady(hermesFrontend).expectStatus().isEqualTo(SERVICE_UNAVAILABLE);
+        getStatusHealth(hermesFrontend).expectStatus().is2xxSuccessful();
+
+        // when
+        kafka.restoreConnectionsBetweenBrokersAndClients();
+
+        // then
+        await().atMost(5, SECONDS)
+                .until(() -> getStatusReady(hermesFrontend).expectStatus().is2xxSuccessful());
+        getStatusHealth(hermesFrontend).expectStatus().is2xxSuccessful();
+
+        // cleanup
+        hermesFrontend.stop();
+    }
+
+    @Test
+    public void shouldRespectKafkaCheckEnabledFlag() {
+        // given
+        kafka.cutOffConnectionsBetweenBrokersAndClients();
+
+        // when
+        HermesTestApp hermesFrontend = new HermesFrontendTestApp(hermesZookeeper, kafka)
+                .metadataMaxAgeInSeconds(1)
+                .readinessCheckIntervalInSeconds(1)
+                .kafkaCheckDisabled()
+                .start();
+
+        // then
+        getStatusReady(hermesFrontend).expectStatus().is2xxSuccessful();
+        getStatusHealth(hermesFrontend).expectStatus().is2xxSuccessful();
+
+        // cleanup
+        kafka.restoreConnectionsBetweenBrokersAndClients();
+        hermesFrontend.stop();
+    }
+
+    @Test
+    public void shouldNotBeReadyUntilThereAreNoUnderReplicatedPartitions() throws Exception {
+        // given
+        List<BrokerId> brokers = kafka.getAllBrokers();
+        List<BrokerId> brokersToStop = brokers.subList(kafka.getMinInSyncReplicas() - 1, brokers.size());
+        kafka.stop(brokersToStop);
+        assertThat(kafka.countUnderReplicatedPartitions() > 0).isTrue();
+
+        // when
+        HermesTestApp hermesFrontend = new HermesFrontendTestApp(hermesZookeeper, kafka)
+                .metadataMaxAgeInSeconds(1)
+                .readinessCheckIntervalInSeconds(1)
+                .kafkaCheckEnabled()
+                .start();
+
+        // then
+        getStatusReady(hermesFrontend).expectStatus().isEqualTo(SERVICE_UNAVAILABLE);
+        getStatusHealth(hermesFrontend).expectStatus().is2xxSuccessful();
+
+        // when
+        kafka.start(selectOne(brokersToStop));
+
+        // then
+        await().atMost(5, SECONDS)
+                .until(() -> getStatusReady(hermesFrontend).expectStatus().is2xxSuccessful());
+        getStatusHealth(hermesFrontend).expectStatus().is2xxSuccessful();
+
+        // cleanup
+        kafka.makeClusterOperational();
```

Comment:
Just sharing my thoughts - I know that this method is not from this PR, but its name is quite general 🤔

Reply:
Do you have any suggestions on how to change it? Would `startAllStoppedBrokers` be better?

Reply:
I'll change it in a separate PR